app/graphql/mutations/attach_files_to_sample.rb

```diff
@@ -13,47 +13,83 @@ class AttachFilesToSample < BaseMutation
              description: 'Persistent Unique Identifier of the sample. For example, `INXT_SAM_AAAAAAAAAA`.'
     validates required: { one_of: %i[sample_id sample_puid] }
 
-    field :errors, GraphQL::Types::JSON, null: false, description: 'Errors that prevented the mutation.'
-    field :sample, Types::SampleType, null: false, description: 'The updated sample.'
-    field :status, GraphQL::Types::JSON, null: false, description: 'The status of the mutation.'
+    field :errors, [Types::UserErrorType], null: false, description: 'A list of errors that prevented the mutation.'
+    field :sample, Types::SampleType, null: true, description: 'The updated sample.'
+    field :status, GraphQL::Types::JSON, null: true, description: 'The status of the mutation.'
 
-    def resolve(args)
+    def resolve(args) # rubocop:disable Metrics/MethodLength,Metrics/AbcSize
       sample = if args[:sample_id]
                  IridaSchema.object_from_id(args[:sample_id], { expected_type: Sample })
                else
                  Sample.find_by(puid: args[:sample_puid])
                end
+
+      if sample.nil?
```

Comment:
**nitpick (blocking):** This error handling when `sample_puid` is provided is different than if a bad `sample_id` is provided. In the case of bad puid you return an errors object in the normal response, but if a bad id is provided the graphql aborts and returns an error.

After you make the `rescue_from` changes for bad GID. You can change `find_by` to `find_by!` and then use a rescue ActiveRecord::RecordNotFound to set `sample` to nil and then this block would have the same behaviour.

Reply:
Changed in https://github.com/phac-nml/irida-next/pull/669/commits/f2a3ab4afd9aefb41993583921f38d5c6095ab29 and https://github.com/phac-nml/irida-next/pull/669/commits/c06ff5bccf93c44e02a02ad3006389726e563f3e